src/Dscp/DB/SQLite/Queries.hs

```diff
@@ -0,0 +1,313 @@
+        insert into  Submissions
+        values       (?, ?, ?, ?, ?)
+    |]
+
+setStudentAssignment :: DBM m => StudentId -> Hash Assignment -> m (StudentId, Hash Assignment)
+setStudentAssignment student assignment = do
+    execute setStudentAssignmentRequest (student, assignment)
+    return (student, assignment)
+  where
+    setStudentAssignmentRequest :: Query
+    setStudentAssignmentRequest = [q|
+        insert into  StudentAssignments
+        values      (?, ?)
+    |]
+
+isAssignedToStudent :: DBM m => StudentId -> Hash Assignment -> m Bool
+isAssignedToStudent student assignment = do
+    exists getStudentAssignmentQuery (student, assignment)
+  where
+    getStudentAssignmentQuery :: Query
+    getStudentAssignmentQuery = [q|
+        select  1
+        from    StudentAssignments
+        where   student_addr = ?
+           and  assignment_hash = ?
+    |]
+
+getGradesForCourseAssignments :: DBM m => StudentId -> CourseId -> m [PrivateTx]
+getGradesForCourseAssignments student course = do
+    query getGradesForCourseAssignmentsQuery (course, student)
+  where
+    getGradesForCourseAssignmentsQuery :: Query
+    getGradesForCourseAssignmentsQuery = [q|
+        select     Submissions.student_addr,
+                   Submissions.contents_hash,
+                   Assignments.course_id,
+                   Assignments.contents_hash,
+                   Assignments.desc,
+                   Submissions.signature
+                   grade
+                   time
+
+        from       Transactions
+
+        left join  Submissions
+               on  submission_hash = Submissions.hash
+
+        left join  Assignments
+               on  sssignment_hash = Assignments.hash
+              and  Assignments.course_id = ?
+
+        where      student_addr = ?
+    |]
+
+getStudentTransactions :: DBM m => PublicKey -> StudentId -> m [PrivateTx]
+getStudentTransactions pk student = do
+    map ($ pk) <$> query getStudentTransactionsQuery (Only student)
+  where
+    getStudentTransactionsQuery :: Query
+    getStudentTransactionsQuery = [q|
+        select     Submissions.student_addr,
+                   Submissions.contents_hash,
+                   Assignments.course_id,
+                   Assignments.contents_hash,
+                   Assignments.desc,
+                   Submissions.signature
+                   grade
+                   time
+
+        from       Transactions
+
+        left join  Submissions
+               on  submission_hash = Submissions.hash
+
+        left join  Assignments
+               on  sssignment_hash = Assignments.hash
+
+        where      student_addr = ?
+    |]
+
+createCourse :: DBM m => CourseId -> Maybe Text -> m CourseId
+createCourse course desc = do
+    execute createCourseRequest (course, desc)
+    return course
+  where
+    createCourseRequest = [q|
+        insert into  Courses
+        values       (?, ?)
+    |]
+
+assert :: DBM m => m Bool -> DomainError -> m ()
```

Comment:
I still think that these should be in `Dscp.Util`.

Reply:
Will move.